### PR TITLE
User/pepijn/2025 03 17 act different image shapes

### PR DESCRIPTION
**Pull request by pkooij**
## What this does
It adapt the forward method in ACT so it works with rotated images or images in datasets that have different shapes but still have the same total pixels (640,480) and (480,640). 

## How it was tested
Runned train.py with dataset that has different image shapes (640, 480) and (480,640)

Command for dataset with rotation:
```bash
python lerobot/scripts/train.py \        
  --dataset.repo_id=pepijn223/act_lekiwi_cam_test2 \ 
  --policy.type=act \
  --output_dir=outputs/train/act_lekiwi_test \                    
  --job_name=act_lekiwi_test \
  --policy.device=mps \                  
  --wandb.enable=true
```

Command for dataset without rotation:
```bash
python lerobot/scripts/train.py \        
  --dataset.repo_id=pepijn223/lekiwi_block_cleanup2 \ 
  --policy.type=act \
  --output_dir=outputs/train/act_lekiwi_test \                    
  --job_name=act_lekiwi_test \
  --policy.device=mps \                  
  --wandb.enable=true
```

### CODE REVIEW

**Review by Cadene — COMMENTED** [review]
What do you think of simplifying by removing the option `batch["observation.images"] = torch.stack(images, dim=-4)`. I dont think it adds anything.

File: lerobot/common/policies/act/modeling_act.py [hunk 1]
```diff
@@ -487,23 +483,24 @@ def forward(self, batch: dict[str, Tensor]) -> tuple[Tensor, tuple[Tensor, Tenso
 
         # Camera observation features and positional embeddings.
         if self.config.image_features:
-            all_cam_features = []
-            all_cam_pos_embeds = []
+            all_tokens = []
+            all_pos_tokens = []
```
Comment (Cadene) [inline lerobot/common/policies/act/modeling_act.py:487]: tokens are supposed to be string/integer ; here it's more features. I would revert the naming

**Review by Cadene — APPROVED** [review]
left 1 comment, besides LGTM
